demos/my_demo_clp.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+
+#include <seqan/arg_parse.h>
+
+struct ModifyStringOptions
+{
+    unsigned period;
+
+    ModifyStringOptions() :
+    period(1)
+    {}
+};
+
+seqan::ArgumentParser::ParseResult
+parseCommandLine(ModifyStringOptions & options, int argc, char const ** argv)
+{
+    // Setup ArgumentParser.
+    //seqan::ArgumentParser parser("FeatureFinderCentroided");
+    seqan::ArgumentParser parser("FeatureFinderCentroided");
+    setVersion(parser, "1.2.3");
+    setUrl(parser, "http://open-ms.sourceforge.net/downloads/");
+
+    // Define Options
+    addOption(parser, seqan::ArgParseOption(
+        "i", "period", "Period to use for the index.",
+        seqan::ArgParseArgument::INTEGER, "INT"));
+    setDefaultValue(parser, "period", "1");
+
+    // Parse command line.
+    seqan::ArgumentParser::ParseResult res = seqan::parse(parser, argc, argv);
+
+    // Only extract  options if the program will continue after parseCommandLine()
+    if (res != seqan::ArgumentParser::PARSE_OK)
+        return res;
+
+    // Extract option values.    
+    getOptionValue(options.period, parser, "period");
+
+    return seqan::ArgumentParser::PARSE_OK;
+}
+
+int main(int argc, char const ** argv)
```

Comment:
Should this entire file go away? Because we cannot test it easily if it requires arguments.